src/main/scala/edu/cmu/cs/obsidian/codegen/yulAST.scala

```diff
@@ -0,0 +1,48 @@
+package edu.cmu.cs.obsidian.codegen
+
+// reminder: use abstract class if want to create a base class that requires constructor arguments
+sealed trait YulAST
+
+object LiteralKind extends Enumeration {
+    type LiteralKind = Value
+    val number, boolean, string = Value
+}
+trait Expression extends YulAST
+trait YulStatement extends YulAST
+
+
+// for each asm struct, create a case class
+case class TypedName(name: String, ntype: String) extends YulAST
+case class Case(value: Literal, body: Block) extends YulAST
+
+case class Literal(kind:LiteralKind.LiteralKind, value: String, vtype: String) extends Expression
+case class Identifier(name: String) extends Expression
+case class FunctionCall(functionName: Identifier, arguements: Seq[Expression]) extends Expression
+
+case class Assignment(variableNames: Seq[Identifier], value: Expression) extends YulStatement
+case class VariableDeclaration(variables: Seq[TypedName]) extends YulStatement
+case class FunctionDefinition(name:String, parameters:Seq[TypedName],
```

Comment:
Either one line per parameter or one line for all parameters. See Scala style guide: https://docs.scala-lang.org/style/indentation.html